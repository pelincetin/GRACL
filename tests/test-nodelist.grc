/*
void printNodeList(NodeList nl) {
    for (Node n in nl) {
        print(n.data());
    }
}
*/

int main()
{
    Graph g = createGraph();

    String hello = "Hello";
    String goodbye = "Goodbye";

    Node n1 = g.createNode(hello);
    Node n2 = g.createNode(goodbye);

    NodeList nl = createNodeList();
    print("Should be true: ");
    if (nl.empty()) {
        print("true");
    } else {
        print("false");
    }

    print("Append node 'hello'");
    nl.appendNode(n1);
    print("Prepend node 'goodbye'");
    nl.prependNode(n2);
    /*
    print("List should be goodbye hello: ");
    printNodeList(nl);
    */

    print("Data of head should be goodbye: ");
    print(nl.head().data());
    print("Data of tail should be hello: ");
    print(nl.tail().data());

    print("Length should be 2: ");
    printi(nl.length());

    nl.removeFirst();
    /*
    print("After remove first (list should be hello): ");
    printNodeList(nl);
    */

    print("Prepend node 'goodbye'");
    nl.prependNode(n2);
    nl.removeLast();
    /*
    print("After remove last (list should be goodbye): ");
    printNodeList(nl);
    */

    print("Append node 'hello'");
    nl.appendNode(n1);
    nl.removeNode(n1);
    /*
    print("After remove node hello (list should be goodbye): ");
    printNodeList(nl); 
    */

    return 0;
}