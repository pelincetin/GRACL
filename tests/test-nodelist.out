Should be true: true
Append node 'hello'
Prepend node 'goodbye'
Data of head should be goodbye: Goodbye
Data of tail should be hello: Hello
Length should be 2: 2
Prepend node 'goodbye'
Append node 'hello'
